scipy/sparse/linalg/_interface.py

```diff
@@ -80,57 +82,90 @@ class LinearOperator:
     ``_adjoint`` is preferable; ``_rmatvec`` is mostly there for
     backwards compatibility.
 
+    The defined operator may have additional "batch" dimensions
+    prepended to the core shape, to represent a batch of 2-D operators;
+    see :ref:`linalg_batch` for details.
+    TODO: check whether we need to add any caveats for broadcasting.
+
     Parameters
     ----------
     shape : tuple
-        Matrix dimensions ``(M, N)``.
+        Matrix dimensions ``(..., M, N)``,
+        where ``...`` represents any additional batch dimensions.
     matvec : callable f(v)
-        Returns returns ``A @ v``.
+        Returns ``A @ v``, where ``v`` is a dense vector
+        with shape ``(..., N)`` or ``(..., N, 1)``.
+        If `shape` contains batch dimensions, this must handle batched input.
```

Comment:
The way this is phrased ("must"), it looks like this adds a new (backward incompatible?) requirement for `LinearOperator` authors?